### DISCUSSION TITLE
Todomvc with Signals?

### DISCUSSION THREAD

**Discussion by ghost** [sycamore-rs/sycamore#523] [General]
Hello.

After spending a week trying to adapt the `Signal` to a real project, I gave up and rewrote everything in `RcSignal` style (as in your ["todomvc"](https://github.com/sycamore-rs/sycamore/blob/master/examples/todomvc/src/main.rs) example).

Is it possible to write something in complexity like todomvc with `Signal`s? First of all, the problem is working with values of type `Signal<Vec<Signal<SomeStruct>>>` (or maybe `Signal<Vec<&'a Signal<SomeStruct>>>`). They throw lifetime and borrow errors, no matter how many `clone()` or `as_ref()` are added. Especially during `set`ting new values or Iteration inside view!{}

Sorry, I'm not a expert on lifetimes in Rust. I would just be happy to have an example of todomvc-like example written on brand new, shiny `Signals`

**Comment by lukechu10** [comment]
Usually the right way to do this would be only to have the outer-most be a `Signal` and everything that is nested be a `RcSignal`. It is technically more correct to use nested `RcSignal` than nested `Signal`s because in the former case, the data is owned whereas in the latter case, the data is just a reference to something allocated on the scope. That means that if you call `create_signal` in a loop, none of it will be dropped until the reactive scope is dropped.
